guides/contributing.md

```diff
@@ -0,0 +1,54 @@
+## Contributing to the Holochain Scaffolding
+
+### Dev environment setup
+
+This project uses the [Holonix](https://developer.holochain.org/get-started/install-advanced/) development environment. Please ensure that you have Nix installed and configured according to that guide. You could develop for the scaffolding without it but that is the environment the project's maintainers use so you may run into build issues we don't know about.
+
+Once you have Nix installed, use a terminal to navigate to a directory where you keep source code and clone this repository:
+
+```bash
+git clone https://github.com/holochain/scaffolding.git
+```
+
+Now enter the development shell:
+
+```bash
+cd scaffolding
+nix develop
+```
+
+You should now see your terminal prompt change to look something like `[rustDev:~/source/holo/scaffolding]$`.
+
+You are now set up, you can start making changes or check the next section to learn how to run the tests.
+
+### Running the scaffolding tests
+
+The tests expect to be able to find the scaffolding CLI in your system path. One easy way to make this to happen is to install it directly using `cargo`:
+
+```bash
+cargo install --path .
+```
+
+You should now be able to run:
+
+```bash
+hc-scaffold --version
+```
+
+and see a version number like `holochain_scaffolding_cli 0.1.11`.
+
+To run the tests, run the provided script:
+
+```bash
+./run_tests.sh
+```
+
```

Comment:
do we want to give the alternative of compiling `hc-scaffold` as it's done on CI?

````suggestion

#### Using `nix develop`
Alternatively you can replicate how CI compiles and introduces `hc-scaffold` to its PATH and run the test script via an ad-hoc nix environment with the following command:

```
nix develop --override-input "versions/scaffolding" . .#ci --command ./run_test.sh`
```
````